micrometer-test/src/main/java/io/micrometer/core/tck/DefaultLongTaskTimerTest.java

```diff
@@ -97,4 +113,24 @@ protected LongTaskTimer newLongTaskTimer(Meter.Id id,
         }
     }
 
+    @Test
+    void histogramToStringNotThrowingException() throws InterruptedException {
+        SimpleMeterRegistry simpleMeterRegistry = new SimpleMeterRegistry();
+
+        LongTaskTimer timer = LongTaskTimer.builder("jobrunr.jobs")
+            .publishPercentiles(0.25, 0.5, 0.75, 0.8, 0.9, 0.95)
```

Comment:
The test fails if this is changed to the other example given in the issue: `0.11, 0.22, 0.33, 0.44`. I'm not sure why it makes a difference to be honest but the following change makes both cases pass without the change to `DefaultLongTaskTimer`.

```diff
Index: micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java b/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java
--- a/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java	(revision e38f5e87dac6566e7909628afa7aa4f7fde6f170)
+++ b/micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/HistogramGauges.java	(date 1705478240207)
@@ -116,7 +116,7 @@
             ToDoubleFunction<HistogramSupport> percentileValueFunction = m -> {
                 snapshotIfNecessary();
                 polledGaugesLatch.countDown();
-                return percentileValue.apply(snapshot.percentileValues()[index]);
+                return percentileValue.apply(valueAtPercentiles[index]);
             };
 
             Gauge.builder(percentileName.apply(valueAtPercentiles[i]), meter, percentileValueFunction)
```

Reply:
My suggested change is not correct. We definitely need more tests in this area of the code to be confident the behavior is correct. I'll take a closer look tomorrow.